### PR TITLE
Axis properties

### PR DESCRIPTION
**Pull request by YuhanLu**
Support more properties for axis

### CODE REVIEW

File: src/compile/axis.ts [hunk 1]
```diff
@@ -195,6 +195,14 @@ export function tickSize(model: Model, channel: Channel) {
   return undefined;
 }
 
+export function tickSizeEnd(model: Model, channel: Channel) {
```
Comment (kanitw) [inline src/compile/axis.ts]: You still haven't modify the top of this file (line 86) as suggested earlier.

File: src/compile/axis.ts [hunk 2]
```diff
@@ -295,6 +326,41 @@ export namespace properties {
       }
     }
 
+    if (axis.tickLabelColor !== undefined) {
```
Comment (kanitw) [inline src/compile/axis.ts]: Why is this method the only method that doesn't use `extend`

Comment (YuhanLu) [inline src/compile/axis.ts]: labelsSpec is extend as well on line 280

Comment (kanitw) [inline src/compile/axis.ts]: @YuhanLu  

Looking closer, it seems like this method here will be throw error if `labelSpec` is `undefined`  (my old code starting from line 287 has the same problem too.  

Can you go through this method and make sure if it's still correct even though the input `labelSpec` is undefined? 

Also, make sure that this method never returns empty object (`{}`).

File: src/compile/axis.ts [hunk 3]
```diff
@@ -295,6 +325,41 @@ export namespace properties {
...
+    }
+
+    if (axis.tickLabelFontSize !== undefined) {
+        labelsSpec.fontSize = {value: axis.tickLabelFontSize};
+    }
+
     return labelsSpec || undefined;
   }
+
+  export function ticks(model: Model, channel: Channel, ticksPropsSpec, def) {
```
Comment (kanitw) [inline src/compile/axis.ts]: @YuhanLu 

let's remove `def` from all of these methods if it's not used.  (If it's used like `labels()`, we can leave it in.)

**Comment by YuhanLu** [comment]
did I press some wrong buttons? why is it closed?

**Comment by YuhanLu** [comment]
All the properties are tested in editor

File: src/compile/axis.ts (hunk 1 above)
Comment (YuhanLu) [inline src/compile/axis.ts]: Done

**Comment by kanitw** [comment]
LGTM.  One last thing.  Can you add documentation for these added properties?

**Comment by YuhanLu** [comment]
Documentation added

File: site/docs/config.md [hunk 4]
```diff
@@ -249,7 +249,7 @@ vg.embed('#horizontal_line', {
 | Property            | Type                | Description  |
 | :------------------ |:-------------------:| :------------|
 | font                | String  | The typeface to set the text in (e.g., `Helvetica Neue`).|
-| fontSize            | Number  | The font size, in pixels. The default value is 10. |
+  | fontSize            | Number  | The font size, in pixels. The default value is 10. |
```
Comment (kanitw) [inline site/docs/config.md]: why are there whitespace in the front of this line?

Comment (YuhanLu) [inline site/docs/config.md]: Sorry that is a typo, just fixed it.

**Comment by domoritz** [comment]
🎉

**Pull request merged**